forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,245 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import csv
+import importlib
+from typing import Iterator, Tuple, List
+from forte.common import Resources, ProcessorConfigError
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.base_reader import PackReader
+from ft.onto.base_ontology import *
+
+
+__all__ = ["ClassificationDatasetReader"]
+
+
+class ClassificationDatasetReader(PackReader):
+    r"""class:`ClassificationDatasetReader` is designed to read text
+    classification dataset that contains input text and digit/text labels.
+
+    User must specify `forte_data_fields` which is a list of forte ontologies
```

Comment:
Try this:

This reader is designed to read table-like classification datasets, there are a couple of values that need to be provided via configuration, including `forte_data_fields`, `input_ontologies`, and `index2class`.

`forte_data_fields` is a list representing the column headers of the dataset.  Each element in the `forte_data_fields` list is either a label that indicates the document class or a Forte entry type that will be used to store the content. Apparently, this list should follow the column order of the target dataset. 

For example:  in the amazon polarity dataset (https://huggingface.co/datasets/amazon_polarity), the columns are [label, title, content]. We can configure `forte_data_fields`  to be ['label', 'ft.onto.base_ontology.Title',
    'ft.onto.ag_news.Description']. `label` is a special keyword to specify the label/document class column, while the latter two ontology types will be used by the reader to store the text from the `title` and `content` column respectively.

Users could also provide `input_ontologies` via configuration, to indicate which of the above-specified fields will be kept. For example, if `title` is not needed, then you can specify this value as [`ftx.onto.ag_news.Description `]. Then only the text from the description part will be stored.